### ISSUE TITLE
Include accessibility validation for buckets with requester-pays configuration.

### ISSUE THREAD

**Issue by sharkinsspatial** [developmentseed/cdk-pgstac#3]
Currently, accessibility validation https://github.com/developmentseed/cdk-pgstac/blob/main/lib/ingestor-api/runtime/src/validators.py#L29 for `requester-pays` buckets will fail on the `HEAD` request without including the flag.  The validator should use the `requester-pays` config by default.

**Comment by alukach** [comment]
Closed by #5
